LibreNMS/Util/IP.php

```diff
@@ -54,6 +54,12 @@ public static function fromHexString($hex, $ignore_errors = false)
 
         $hex = str_replace([':', '.'], '', $hex);
 
+        // check if hex was incorrectly converted to ascii
+        $len = strlen($hex);
+        if (($len == 4 || $len == 16) && preg_match('/[^0-9a-fA-F]/', $hex)) {
```

Comment:
Do we have some tests for this? It’s really hard to understand why it’s needed else, or what the length check is

Reply:
Juniper is dumb.

From the linked issue: https://supportportal.juniper.net/s/article/Junos-Values-for-show-snmp-mib-walk-jnxBgpM2PeerLocalAddr-don-t-appear-correct?language=en_US

Basically any hex in the ascii range gets converted to ascii.  Because IPs are fixed length, IPv4 will be 4 characters and IPv6 will be 16 characters.  The length check is to avoid running the preg_match every time and improve the accuracy of the check.

Reply:
Tests would be a good idea ;)